File: Algorithms/permutation-palindrome.js
//Write an efficient function that checks whether any permutation ↴ of an input string is a palindrome ↴ .
function palinPermutation(str){
  var strMap = {};

  for (var i = 0; i < str.length; i++){
    if (strMap[str[i]]){
      strMap[str[i]] = false;
    } else {
      strMap[str[i]] = true;
    }
  }

  var odds = 0;

  for (var letter in strMap){
    if (strMap[letter] === true){
      odds++;
    }
  } 
  return odds === 1
}

File: Algorithms/Ruby/permutation-palindrome.rb
def permutation_palindrome(str)
  dict = {}
  odds = 0

  str.chars.map do |val|
   dict[val] = (dict[val]) ? dict[val] += 1 : 1
  end

  dict.keys.map do |key|
    odds += 1 if dict[key] % 2 != 0
  end

  return odds <= 1
end